src/azure-cli/azure/cli/command_modules/appconfig/_snapshot_custom_client.py

```diff
@@ -0,0 +1,595 @@
+from azure.appconfiguration._generated.models import Error as AppConfigError
+from azure.core.rest import HttpRequest
+from azure.core.paging import ItemPaged
+from azure.core.utils import case_insensitive_dict
+from azure.core.exceptions import ClientAuthenticationError, ResourceExistsError, ResourceNotFoundError, HttpResponseError, ResourceModifiedError, map_error
+from msrest import Serializer
+import json
+
+from ._constants import SnapshotConstants
+from ._snapshotmodels import Snapshot, SnapshotListResult, SnapshotCreateData
+
+
+class ProvisioningStatus:
+    RUNNING = "Running"
+    SUCCEEDED = "Succeeded"
+    FAILED = "Failed"
+
+
+class RequestMethod:
+    GET = "GET"
+    PUT = "PUT"
+    PATCH = "PATCH"
+    DELETE = "DELETE"
+
+
+_ERROR_MAP = {
+    401: ClientAuthenticationError,
+    404: ResourceNotFoundError,
+    409: ResourceExistsError,
+    412: ResourceModifiedError
+}
+
+
+def _build_request(
+    template_url_default,
+    method,
+    *,
+    path_arguments=None,
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+    _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+    _params = case_insensitive_dict(kwargs.pop("params", {}) or {})
+
+    api_version = kwargs.pop("api_version", SnapshotConstants.API_VERSION)
+    accept = _headers.pop(
+        "Accept", "application/json, application/problem+json"
+    )
+
+    # Construct URL
+    _url = kwargs.pop("template_url", template_url_default)
+    _path_arguments = path_arguments or {}
+    _url = _format_url_section(_url, **_path_arguments)
+
+    _serializer = Serializer()
+
+    # Construct parameters
+    _params["api-version"] = _serializer.query("api_version", api_version, "str")
+
+    # Construct headers
+    if sync_token is not None:
+        _headers["Sync-Token"] = _serializer.header("sync_token", sync_token, "str")
+
+    if if_match is not None:
+        _headers["If-Match"] = _serializer.header("if_match", if_match, "str")
+    if if_none_match is not None:
+        _headers["If-None-Match"] = _serializer.header("if_none_match", if_none_match, "str")
+
+    _headers["Accept"] = _serializer.header("accept", accept, "str")
+
+    return HttpRequest(method=method.upper(), url=_url, params=_params, headers=_headers, **kwargs)
+
+
+# Requests
+def build_get_snapshot_request(
+    name,
+    select=None,
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+    _params = {}
+
+    if select is not None:
+        _params["$Select"] = Serializer().query("select", select, "[str]", div=",")
+
+    return _build_request(
+        "/snapshots/{name}",
+        RequestMethod.GET,
+        path_arguments={"name": name},
+        if_match=if_match,
+        if_none_match=if_none_match,
+        sync_token=sync_token,
+        params=_params,
+        **kwargs
+    )
+
+
+def build_get_snapshot_create_status(
```

Comment:
Ideally we should have a generic "get operation status" method.

Reply:
If we have to special case things for snapshots, then it may indicate that we didn't properly implement the long-running-operation model.

Reply:
he implementations for these are embedded in individual classes and there are no standalone utility functions for this. The implementation was therefore based on the method outlined [here](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/polling/base_polling.py#:~:text=def%20request_status(self%2C%20status_link)%3A)